src/com/com_nrf24l01_core.cpp

```diff
@@ -0,0 +1,312 @@
+    config_register.ClearBit(reg::config::PWR_UP);
+  }
+
+  return (spi_protocol_->WriteRegister(reg::config::REG_ADDR, config_register.Get()));
+}
+
+auto NRF24L01Core::EnableDataPipe(DataPipe pipe_no) noexcept -> types::DriverStatus {
+  auto get_en_rxaddr_reg = spi_protocol_->ReadRegister(reg::en_rxaddr::REG_ADDR);
+  if (get_en_rxaddr_reg.first != types::DriverStatus::OK) {
+    return get_en_rxaddr_reg.first;
+  }
+  utilities::Byte en_rxaddr_reg(get_en_rxaddr_reg.second);
+  en_rxaddr_reg.SetBit(static_cast<uint8_t>(pipe_no));
+  return (spi_protocol_->WriteRegister(reg::en_rxaddr::REG_ADDR, en_rxaddr_reg.Get()));
+}
+
+auto NRF24L01Core::DisableDataPipe(DataPipe pipe_no) noexcept -> types::DriverStatus {
+  auto get_en_rxaddr_reg = spi_protocol_->ReadRegister(reg::en_rxaddr::REG_ADDR);
+  if (get_en_rxaddr_reg.first != types::DriverStatus::OK) {
+    return get_en_rxaddr_reg.first;
+  }
+  utilities::Byte en_rxaddr_reg(get_en_rxaddr_reg.second);
+  en_rxaddr_reg.ClearBit(static_cast<uint8_t>(pipe_no));
+  return (spi_protocol_->WriteRegister(reg::en_rxaddr::REG_ADDR, en_rxaddr_reg.Get()));
+}
+
+auto NRF24L01Core::EnableAutoAck(DataPipe pipe_no) noexcept -> types::DriverStatus {
+  auto get_en_aa_reg = spi_protocol_->ReadRegister(reg::en_aa::REG_ADDR);
+  if (get_en_aa_reg.first != types::DriverStatus::OK) {
+    return get_en_aa_reg.first;
+  }
+  utilities::Byte en_aa_reg(get_en_aa_reg.second);
+  en_aa_reg.SetBit(static_cast<uint8_t>(pipe_no));
+  return (spi_protocol_->WriteRegister(reg::en_aa::REG_ADDR, en_aa_reg.Get()));
+}
+
+auto NRF24L01Core::ConfigAutoRetransmission(AutoRetransmissionDelay delay, AutoRetransmitCount count) noexcept -> types::DriverStatus {
+  auto get_setup_retr_reg = spi_protocol_->ReadRegister(reg::setup_retr::REG_ADDR);
+  if (get_setup_retr_reg.first != types::DriverStatus::OK) {
+    return get_setup_retr_reg.first;
+  }
+  utilities::Byte setup_retr_reg(get_setup_retr_reg.second);
+  setup_retr_reg.SetLowNibble(static_cast<uint8_t>(count));
+  setup_retr_reg.SetHighNibble(static_cast<uint8_t>(delay));
+  return (spi_protocol_->WriteRegister(reg::setup_retr::REG_ADDR, setup_retr_reg.Get()));
+}
+
+auto NRF24L01Core::MaskInterruptOnIntPin(MaskeableInterrupts interrupt) -> types::DriverStatus {
+  auto get_config_register = spi_protocol_->ReadRegister(reg::config::REG_ADDR);
+
+  ON_ERROR_RETURN(get_config_register.first);
+
+  utilities::Byte config_register(get_config_register.second);
+  config_register.SetBit(static_cast<std::uint8_t>(interrupt));
+
+  return (spi_protocol_->WriteRegister(reg::config::REG_ADDR, config_register.Get()));
+}
+
+auto NRF24L01Core::SetPipeAddress(DataPipe pipe_no, data_pipe_address pipe_addr) noexcept -> types::DriverStatus {
+  uint8_t register_addr = reg::rx_addr_p0::REG_ADDR + static_cast<std::uint8_t>(pipe_no);
+  std::vector<std::uint8_t> multibyte_addr;
+  std::uint8_t singlebyte_addr = 0;
+  auto return_value = types::DriverStatus::INPUT_ERROR;
+
+  switch (pipe_no) {
```

Comment:
This is a more complicated single if single else statement. I think it would be cleaner to write it as such.

Reply:
The switch case structure is the more defensive approach. In case pipe_no is corrupt the calculated register is not correct, the content will be overwritten with a wrong value and the function will return an "OK".

Reply:
ok